### PR TITLE
 Support basic auth secret 

### LINKED ISSUE #1103: Add option for creating secret in "namespace init" supporting basic auth of any registry
We currently only support basic auth credentials for dockerhub as part of `namespace init`

We should add an option to do this for any docker registry that uses basic auth.

I suggest adding:
   `riff namespace init default --registry-user someone --registry-server registry.example.com`

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #1103.

### CODE REVIEW

File: pkg/core/namespace.go [hunk 1]
```diff
@@ -210,6 +217,15 @@ func (kc *kubectlClient) createGcrSecret(options NamespaceInitOptions, labels ma
 	return kc.createBasicAuthSecret(options.NamespaceName, options.SecretName, "_json_key", string(token), "https://gcr.io", labels)
 }
 
+func (kc *kubectlClient) createRegistrySecret(options NamespaceInitOptions, labels map[string]string) error {
+	username := options.RegistryUser
+	password, err := readPassword(fmt.Sprintf("Enter password for user %q", username))
+	if err != nil {
+		return err
+	}
+	return kc.createBasicAuthSecret(options.NamespaceName, options.SecretName, username, password, options.RegistryServer, labels)
```
Comment (trisberg) [inline pkg/core/namespace.go]: should we add th eprotocol `https://` if it is not provided? Seems all example here use a protocol - https://github.com/knative/docs/blob/master/build/auth.md#guiding-credential-selection

Comment (fbiville) [reply pkg/core/namespace.go]: or should we reject it maybe? We could parse the URL and reject anything that does not have a http or https protocol?

Comment (trisberg) [reply pkg/core/namespace.go]: should we change the flag to be `--repository-url` then?

Comment (fbiville) [reply pkg/core/namespace.go]: `repository-url` or `registry-url`?

**Comment by ericbottard** [comment]
I dont think this is a good Idea as typically an image coordinate does not
look like a URL

Le lun. 18 févr. 2019 à 18:25, Florent Biville <notifications@github.com> a
écrit :

File: pkg/core/namespace.go (hunk 1 above)
Comment (trisberg) [reply pkg/core/namespace.go]: sorry, meant registry-url, although I would prefer to keep registry-host and just add the protocol

**Comment by trisberg** [comment]
should we also require `--registry-default-path` in order not to conflict with #1105 

Then we could treat 

  `--dockerhub trisberg`

to be equivalent to

  `--registry-host index.docker.io`+`--registry-username trisberg`+`--registry-default-path trisberg`

**Comment by trisberg** [comment]
and we could also add `--registry-protocol` that defaults to `https` but can be overridden for `http`  registries

**Comment by fbiville** [comment]
Added the protocol option.
I just noticed that if I initialize a namespace with the registry host `index.docker.io/v1`, I get 401s when trying to create a function. It works fine with `index.docker.io/v1/`. That's quite strange.

**Comment by fbiville** [comment]
I'd rather wait for #1105 to be merged before addressing the addition of `--registry-default-path`.

**Comment by ericbottard** [comment]
LGTM, successfully tested against docker.io.
The flag validation logic seems legit too

**Comment by ericbottard** [comment]
Merged as 7984f036

**Pull request closed without merging**